### PR TITLE
fix: upgrade gke module to avoid TPGv5.44.0 with Autopilot

### PR DESCRIPTION
**Pull request by genlu2011**
Upgrade gke submoduel version to patch the fix in latest release
https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/releases/tag/v33.0.0

### CODE REVIEW

File: infrastructure/versions.tf [hunk 1]
```diff
@@ -18,7 +18,8 @@ terraform {
       source = "hashicorp/google"
     }
     google-beta = {
-      source = "hashicorp/google-beta"
+      source  = "hashicorp/google-beta"
+      version = ">= 5.40.0, != 5.44.0, !=6.2.0"
```
Comment (andrewsykim) [inline infrastructure/versions.tf:22]: Leave a comment for while these versions are excluded

**Review by andrewsykim — APPROVED** [review]
Minor nit about leaving a comment, otherwise LGTM

File: infrastructure/versions.tf (hunk 1 above)
Comment (genlu2011) [reply infrastructure/versions.tf]: Added

**Pull request merged**